dali/pipeline/operators/crop/crop_attr.h

```diff
@@ -56,45 +57,52 @@ class CropAttr {
         "`crop` argument is not compatible with `crop_h`, `crop_w`, `crop_d`");
 
       auto cropArg = spec.GetRepeatedArgument<float>("crop");
-      DALI_ENFORCE(cropArg.size() > 0 && cropArg.size() <= 2);
-      crop_h = static_cast<int>(cropArg[0]);
-      crop_w = static_cast<int>(cropArg.size() >= 2 ? cropArg[1] : cropArg[0]);
-
-      DALI_ENFORCE(crop_h >= 0,
-        "Crop height must be greater than zero. Received: " +
-        std::to_string(crop_h));
+      if (cropArg.size() == 1) {
+          DALI_WARN("Warning: Single value argument for `crop` is now deprecated. "
+                    "To produce a squared cropping window, please provide both values "
+                    "explicitly `crop=(c, c)`."
+                    "Future releases will treat this as an error");
+          cropArg.push_back(cropArg[0]);
+      }
+      size_t crop_ndims = cropArg.size();
+      DALI_ENFORCE(crop_ndims <= 3,
+        "Cropping windows with more than 3 dimensions are not supported");
 
-      DALI_ENFORCE(crop_w >= 0,
-        "Crop width must be greater than zero. Received: " +
-         std::to_string(crop_w));
+      size_t idx = 0;
+      if (crop_ndims == 3) {
+        crop_d = static_cast<int>(cropArg[idx++]);
+      }
+      crop_h = static_cast<int>(cropArg[idx++]);
+      crop_w = static_cast<int>(cropArg[idx++]);
     }
+    has_crop_d_ = has_crop_d_arg || crop_d > 0;
 
     crop_height_.resize(batch_size__, crop_h);
     crop_width_.resize(batch_size__, crop_w);
-    if (has_crop_d_arg)
+    if (has_crop_d_)
       crop_depth_.resize(batch_size__, crop_d);
     crop_x_norm_.resize(batch_size__, 0.0f);
     crop_y_norm_.resize(batch_size__, 0.0f);
-    if (has_crop_d_arg)
+    if (has_crop_d_)
       crop_z_norm_.resize(batch_size__, 0.0f);
     crop_window_generators_.resize(batch_size__, {});
   }
 
   void ProcessArguments(const ArgumentWorkspace *ws, std::size_t data_idx) {
     crop_x_norm_[data_idx] = spec__.GetArgument<float>("crop_pos_x", ws, data_idx);
     crop_y_norm_[data_idx] = spec__.GetArgument<float>("crop_pos_y", ws, data_idx);
-    if (Is3dCrop())
+    if (has_crop_d_)
       crop_z_norm_[data_idx] = spec__.GetArgument<float>("crop_pos_z", ws, data_idx);
     if (!is_whole_image_) {
-      if (crop_width_[data_idx] == 0) {
+      if (crop_width_[data_idx] == kUnspecifiedCrop) {
```

Comment:
I think the logic is a bit convoluted here and I would prefer to check explicitly if we have `has_crop_w_arg` instead of relying on the fact that we didn't propagate a value from `crop` argument so we should have this one.

Reply:
I'm not sure, but I think we _migh_ have propagated the value of "NoCrop"

Reply:
I mean, we set it first to `kUnspecifiedCrop`, and we try to read the `crop` argument and do the `crop_width_.resize(batch_size__, crop_w);` based on it. If we don't do it, but this `is_whole_image_` check is true, than we know that we can read `crop_w`. I spent like 15 minutes trying to get this.